src/main/java/com/conveyal/datatools/manager/jobs/MonitorServerStatusJob.java

```diff
@@ -248,14 +258,37 @@ private void waitAndCheckInstanceHealth(String waitingFor) throws InstanceHealth
         checkInstanceHealth();
     }
 
+    /**
+     * Helper method to initiate to first attempt at checking the instance health.
+     */
+    private void checkInstanceHealth() throws InstanceHealthException {
+        checkInstanceHealth(1);
+    }
+
     /**
      * Checks whether the instance is running. If it has entered a state where it is stopped, terminated or about to be
      * stopped or terminated, then this method throws an exception.
      */
-    private void checkInstanceHealth() throws InstanceHealthException {
+    private void checkInstanceHealth(int attemptNumber) throws InstanceHealthException {
         DescribeInstancesRequest request = new DescribeInstancesRequest()
             .withInstanceIds(Collections.singletonList(instance.getInstanceId()));
-        DescribeInstancesResult result = ec2.describeInstances(request);
+        DescribeInstancesResult result;
+        try {
+            result = ec2.describeInstances(request);
+        } catch (AmazonEC2Exception e) {
+            LOG.error("Failed on attempt {} to execute request to obtain instance health!", attemptNumber, e);
+            if (attemptNumber > MAX_INSTANCE_HEALTH_RETRIES) {
```

Comment:
Does this need to be a final int? I don't love that it is used only once and defined so far away from its use. Maybe instead the method could just accept a max retries value as one of the args?